### PR TITLE
Versioning of sub parameters

### PR DESCRIPTION
**Pull request by xgreenx**
Wrapped `FeeParameters`, `PredicateParameters`, `TxParameters`, `ScriptParameters` and `ContractParameters` into an enum to support versioning.

### CODE REVIEW

**Review by xgreenx — COMMENTED** [review]
Ignore JS files please, it is auto formatter=)

File: fuel-tx/src/transaction/consensus_parameters.rs [hunk 1]
```diff
@@ -320,50 +467,156 @@ pub struct PredicateParameters {
     pub max_gas_per_predicate: u64,
 }
 
-impl PredicateParameters {
-    /// Default consensus parameters with settings suggested in fuel-specs
+impl PredicateParametersV1 {
+    /// Default parameters just for testing.
```
Comment (Dentosal) [inline fuel-tx/src/transaction/consensus_parameters.rs:471]: If these are just for testing, shouldn't we feature-gate them?

Comment (xgreenx) [reply fuel-tx/src/transaction/consensus_parameters.rs:471]: I will put them under `test-helpers` in a separate PR

**Review by Dentosal — APPROVED** [review]
This is ok as-is, but needs follow-up PR as discussed above.

**Pull request merged**